### DISCUSSION TITLE
deep (too deep !) dive in Nos3

### DISCUSSION THREAD

**Discussion by girardja** [nasa/nos3#281] [Q&A]
Hy, I am a French Engineer, and I have installed the STF1 example through Vagrant.
I can make and make launch.
I have read many documents on nos3, CFS, CFE and many other nitty gritty items for three weeks. I have also read many lines of codes...
I am still confuse about the global picture. 
I would like to understand for example how to upload a new file (let's say a .so) from ground to board
    Is there a command to do that from cosmos interface ? I have seen that (CCSDS File Delivery Protocol) was implemented in CI/TO apps.
    I guess OSAL would store the file somewhere in its filesystem. Can I check that it has been loaded with a ‘ls’ on a directory (maybe fsw/build/exe/cpu1/cf or data  ?)
    How can I request the board to restart the application(s) depending on the (.so) file ?

I have also made many attempts to try to understand by suppressing files from the build tree but I was confuse because the simulator seemed to work as well. For example before make launch, I suppressed all the generic*.so from fsw/build/exe/cpu1/cf)

I also tried to comment many lines in the startup.scr file and here again it didn't seem to have an impact.

I also wonder why the tables and other data are all defined by c code instead of data files (json, xml or whatever). Thus they require compilation to get object and then efl2tbl tool to be written in a tbl format.

So as you see I am a bit lost and your answers will probably help me save time.

**Comment by jlucas9** [comment]
Hi @girardja,
We're starting to develop scenarios and tutorials for new users finally! We plan to demonstrate nominal operations as you mentioned above in issue https://github.com/nasa/nos3/issues/330. A lot of the other oddities are really related to the underlying frameworks we're leveraging in NOS3. For example cFS has been around for many years as an open-source option for flight software and is still being developed and improved to resolve for example the table issues you mentioned.
